### PR TITLE
Fixes #2488 Report Reasons not sent in report PM/email notifications

### LINKED ISSUE #2488: Report Reasons not sent in report PM/email notifications
https://community.mybb.com/thread-205425-post-1250421.html#pid1250421

The reason `title` should be fetched from the `reportreasons` table and the `reason` field should also be included for reasons that require additional information. This should then be inserted [here](https://github.com/mybb/mybb/blob/mybb_1808/inc/functions_modcp.php#L213).

---

While confirming this issue I have also noticed that the `reportmethod` setting should probably be clarified as to what it does.
Old Title: `Reported Posts Medium`
Old Description: `Please select from the list how you want reported posts to be dealt with. Storing them in the database is probably the better of the options listed.`
Old optionscode:

```
radio
db=Stored in the Database
pms=Sent as Private Messages
email=Sent via Email
```

New Title: `Reported Content Notification Method`
New Description: `Please select from the list how you would like moderators to be notified when content is reported. Note: Moderators are only notified for posts in forums they moderate or all content if they are a super moderator.`
New optionscode:

```
radio
db=Shown in the Mod CP only
pms=Sent as Private Messages and shown in the Mod CP
email=Sent via Email and shown in the Mod CP
```
#1223

### PR DESCRIPTION
**Pull request by JordanMussi**

### CODE REVIEW

**Comment by JN-Jones** [comment]
Looks fine to me, however would be nice if someone else takes a second look @mybb/sqa

File: inc/functions_modcp.php [hunk 1]
```diff
@@ -145,6 +145,22 @@ function send_report($report, $report_type='post')
 {
 	global $db, $lang, $forum, $mybb, $post, $thread, $reputation, $user;
 
+	$report_reason = '';
+	if($report['reasonid'])
+	{
+		$query = $db->simple_select("reportreasons", "title", "rid = '{$report['reasonid']}'", array('limit' => 1));
```
Comment (Devilshakerz) [inline inc/functions_modcp.php]: Can you cast `$report['reasonid']` to integer here? It comes from an `int` field in the database, but it's cast nonetheless in other places.

Comment (JN-Jones) [inline inc/functions_modcp.php]: Though I'm against useless casts I'd prefer it here too as it's a function which may be used by plugins

Comment (JN-Jones) [inline inc/functions_modcp.php]: @JordanMussi Can you change this?

Comment (dequeues) [inline inc/functions_modcp.php]: If Jordan is not available we could just push this and then do another commit to cast it as int.

Comment (Ben-Conway) [inline inc/functions_modcp.php]: @JN-Jones Can we get this looked at and possibly in 1.8.9?

Comment (JN-Jones) [inline inc/functions_modcp.php]: Apart from this it looked fine before, need to retest though

**Comment by JN-Jones** [comment]
Seems to work fine, creating a new PR with the fix for the issue mentioned above.
